src/software/new_geom/linear_spline2d.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <vector>
+
+#include "software/new_geom/point.h"
+#include "software/new_geom/polynomial2d.h"
+#include "software/new_geom/spline2d.h"
+
+/**
+ * A linear spline is a spline composed of a series of linear 2d polynomials, ie. a series
+ * of connected line segments
+ */
+class LinearSpline2d : Spline2d
+{
+   public:
+    LinearSpline2d() = delete;
+    /**
+     * Construct a spline by drawing line segments between consecutive
+     * Points
+     *
+     * @throws std::invalid_argument if points.size() == 0
+     *
+     * @param points Points on the spline
+     */
+    explicit LinearSpline2d(const std::vector<Point>& points);
+
+    /**
+     * Construct a spline by drawing line segments between consecutive
+     * Points
+     *
+     * @throws std::invalid_argument if points.size() == 0
+     *
+     * @param points Points on the spline
+     */
+    LinearSpline2d(const std::initializer_list<Point>& points);
+
+    /**
+     * Please see parent class for jdoc
```

Comment:
We already have the convention of not javadocing functions marked `override`, so I don't think we need these javadoc comments for these functions

Reply:
removed all